### PR TITLE
Add permission seeding

### PR DESCRIPTION
**Pull request by astrozzc**
Seed the permission from directory 'permissions'. Also provide
option to seed the permissions manually.

## Link(s) to Jira
https://projects.engineering.redhat.com/browse/RHCLOUD-6573

## Description of Intent of Change(s)
Seed the permissions which will be available to be added to the Access objects.

## Local Testing
How can the feature be exercised? 
Unit test
Is any special local setup required?
No

## Checklist
- [ ] if API spec changes are required, is the spec updated?

- [ ] are there any pre/post merge actions required? if so, document here.

- [x] are theses changes covered by unit tests?

- [x] if warranted, are documentation changes accounted for?

- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?

- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

### CODE REVIEW

File: rbac/management/management/commands/seeds.py [hunk 1]
```diff
@@ -30,17 +30,23 @@ class Command(BaseCommand):
...
     def add_arguments(self, parser):
         """Add arguments to command."""
+        parser.add_argument("--permissions-only", action="store_true")
         parser.add_argument("--roles-only", action="store_true")
         parser.add_argument("--groups-only", action="store_true")
 
     def handle(self, *args, **options):
         """Handle method for command."""
-        if not options["groups_only"]:
+        if options["permissions_only"]:
```
[the remark this answers is not in the document]
Comment (astrozzc) [reply rbac/management/management/commands/seeds.py]: Good catch! Yea, I am changing it to `--roles|--groups|--permissions|--all`, this way we could seed any combinations of them, e.g. both `--groups --permissions`. And seed all of them by `--all`.

File: rbac/management/role/definer.py [hunk 2]
```diff
@@ -84,3 +84,36 @@ def seed_roles(tenant, update=False):
...
+
+    with tenant_context(tenant):
+        for permission_file_name in permission_files:
+            permission_file_path = os.path.join(permission_directory, permission_file_name)
+            app_name = os.path.splitext(permission_file_name)[0]
+            with open(permission_file_path) as json_file:
+                data = json.load(json_file)
+                for resource, operations in data.items():
+                    for operation in operations:
+                        permission, created = Permission.objects.get_or_create(
```
Comment (coderbydesign) [inline rbac/management/role/definer.py:107]: I think we may be want to do something here like an `update_or_create` [1] to ensure we're getting all fields set? It seems this migration [2] is only setting the `permission` field on `Permission`, so we end up with `Permission` records without the other fields populated. 

[1] https://docs.djangoproject.com/en/dev/ref/models/querysets/#update-or-create
[2] https://github.com/RedHatInsights/insights-rbac/pull/291/files#diff-abf50ea78e7ec39af811f223d8c6aa53R15

File: rbac/management/management/commands/seeds.py [hunk 3]
```diff
@@ -18,29 +18,36 @@
...
-        parser.add_argument("--groups-only", action="store_true")
+        parser.add_argument("--permissions", action="store_true")
+        parser.add_argument("--roles", action="store_true")
+        parser.add_argument("--groups", action="store_true")
+        parser.add_argument("--all", action="store_true")
 
     def handle(self, *args, **options):
         """Handle method for command."""
-        if not options["groups_only"]:
+        if options["permissions"] | options["all"]:
```
Comment (coderbydesign) [inline rbac/management/management/commands/seeds.py:40]: Is there an explicit reason to use a bitwise operator here?

File: rbac/management/role/definer.py [hunk 4]
```diff
@@ -84,3 +84,36 @@ def seed_roles(tenant, update=False):
...
+    """For a tenant update or create defined permissions."""
+    permission_directory = os.path.join(settings.BASE_DIR, "management", "role", "permissions")
+    permission_files = [
+        f
+        for f in os.listdir(permission_directory)
+        if os.path.isfile(os.path.join(permission_directory, f)) and f.endswith(".json")
+    ]
+    # current_permission_ids = set()
+
+    with tenant_context(tenant):
```
Comment (coderbydesign) [inline rbac/management/role/definer.py:99]: I think we'll want to use `with transaction.atomic():` here?

File: rbac/management/management/commands/seeds.py (hunk 3 above)
Comment (astrozzc) [reply rbac/management/management/commands/seeds.py:40]: Yea, to provide a convenient way "all" to support do all the seeding so we don't have to explicitly specify roles, groups, and permissions.

File: rbac/management/role/definer.py (hunk 4 above)
Comment (astrozzc) [reply rbac/management/role/definer.py:99]: Ah ha, good idea 👍

File: rbac/management/management/commands/seeds.py (hunk 3 above)
Comment (astrozzc) [reply rbac/management/management/commands/seeds.py]: Sorry, I miss understood, updated 👍

**Pull request merged**